src/requirements/reqs-functions.ts

```diff
@@ -303,9 +306,16 @@ export function computeRequirements(
         .flat();
     },
     getCorrespondingRequirementAndAllRelevantCoursesUnderFulfillmentStrategy: it => it,
-    // TODO: Replace this dummy implementation once we decided how to determine if a requirement is
-    // double-countable. Meanwhile, make it always return true to match the old behavior.
-    allowDoubleCounting: () => true,
+    allowDoubleCounting: requirement => {
+      // All minor requirements are automatically double-countable.
+      if (requirement.sourceType === 'Minor') return true;
+      if (requirement.sourceType === 'Major') {
+        if (majors == null) throw new Error();
+        // If it's not the first major, then it's double countable.
+        if (requirement.sourceSpecificName !== majors[0]) return true;
```

Comment:
Just want to clarify, does this mean that anything can be double counted by the second major that fulfills a requirement in the first major, or does it also mean that requirements in the second major are double countable with each other?

Reply:
short answer: all the non-first majors are treated the same as minors.
then all the minor requirements allow double counting automatically. e.g. if a course A satisfy a minor requirement, it can still be used to satisfy other requirements that doesn't allow double counting.